src/main/java/org/jenkins/tools/test/hook/MultiParentCompileHook.java

```diff
@@ -114,53 +113,43 @@ private void copy(Path eslintrc, File pluginFolder) {
         }
     }
 
-    private MavenRunner.Config getMavenConfig(PluginCompatTesterConfig config) {
-        MavenRunner.Config mconfig = new MavenRunner.Config(config);
-        // TODO REMOVE
-        mconfig.userProperties.put("failIfNoTests", "false");
-        return mconfig;
-    }
-
-    private void compile(
-            MavenRunner.Config mavenConfig,
-            File path,
-            File localCheckoutDir,
-            String parentFolder,
-            String pluginName)
+    private void compile(File path, File localCheckoutDir, String parentFolder, String pluginName)
             throws PomExecutionException {
         if (isSnapshotMultiParentPlugin(parentFolder, path, localCheckoutDir)) {
             // "process-test-classes" not working properly on multi-module plugin.
             // See https://issues.jenkins.io/browse/JENKINS-62658
             // installs dependencies into local repository
-            String mavenModule =
-                    PluginCompatTester.getMavenModule(pluginName, path, runner, mavenConfig);
+            String mavenModule = PluginCompatTester.getMavenModule(pluginName, path, runner);
             if (mavenModule == null || mavenModule.isBlank()) {
                 throw new IllegalStateException(
                         String.format(
                                 "Unable to retrieve the Maven module for plugin %s on %s",
                                 pluginName, path));
             }
             runner.run(
-                    mavenConfig,
+                    Map.of(
+                            "skipTests",
+                            "true",
+                            "invoker.skip",
+                            "true",
+                            "enforcer.skip",
+                            "true",
+                            "maven.javadoc.skip",
+                            "true"),
                     path.getParentFile(),
                     setupCompileResources(path.getParentFile()),
                     "clean",
                     "install",
-                    "-DskipTests",
-                    "-Dinvoker.skip",
-                    "-Denforcer.skip",
-                    "-Dmaven.javadoc.skip",
                     "-am",
                     "-pl",
                     mavenModule);
         } else {
             runner.run(
-                    mavenConfig,
+                    Map.of("maven.javadoc.skip", "true"),
                     path,
                     setupCompileResources(path),
                     "clean",
-                    "process-test-classes",
-                    "-Dmaven.javadoc.skip");
+                    "process-test-classes");
```

Comment:
So this is a call to Maven for compilation. `mavenConfig` previously had `-DfailIfNoTests=false` which was irrelevant in this context. Here I have moved the `-D` arguments from the varargs into the first parameter for clarity. I guess you should be noticing a pattern by now: when running Maven for compilation, we pretty much always want to be skipping Javadoc generation.